lib/index.js

```diff
@@ -33,11 +33,11 @@ export function getKeys(node) {
 // eslint-disable-next-line valid-jsdoc
 /**
  * Make the union set with `KEYS` and given keys.
- * @param {Object} additionalKeys The additional keys.
- * @returns {{ [type: string]: string[] | undefined }} The union set.
+ * @param {{ readonly [type: string]: ReadonlyArray<string> }} additionalKeys The additional keys.
+ * @returns {{ readonly [type: string]: ReadonlyArray<string> }} The union set.
  */
 export function unionWith(additionalKeys) {
-    const retv = Object.assign({}, KEYS);
+    const retv = /** @type {{ [type: string]: ReadonlyArray<string> }} */ (Object.assign({}, KEYS));
```

Comment:
Can we name this type to avoid inlining long declarations like this?

Reply:
Sure, I've added a commit. Just note that this adds marginally to the declaration file, as adding a `@typedef` results in new `type`s being created (though that may actually be a convenience for users).